doc/developer/design/20230607_shift_status_updates.md

```diff
@@ -0,0 +1,151 @@
+# Shift Storage Status Updates
+
+- [Shift Storage Status Updates](#shift-storage-status-updates)
+  - [Associated](#associated)
+  - [Context](#context)
+  - [Goals](#goals)
+  - [Overview](#overview)
+  - [Detailed description](#detailed-description)
+    - [Rollout strategy](#rollout-strategy)
+  - [Alternatives](#alternatives)
+  - [Open questions](#open-questions)
+
+
+## Associated
+
+- [#18795](https://github.com/MaterializeInc/materialize/issues/18795)
+
+<!--
+Note: Feel free to add or remove sections as needed. However, most design
+docs should at least keep the suggested sections.
+-->
+
+## Context
+
+<!--
+Bring the reader up to speed enough, such that they can understand the
+following goals and descriptions.
+
+An important reason for this is helping future readers understand the
+assumptions that went into the design, and in turn the goals and design itself.
+
+Be sure to capture the customer impact/customer problem, which should be the
+motivation for the proposed design!
+-->
+
+Presently, status updates for source ingestion and sink dataflows are written to the dataflow's designated persist status shard. These writes happen periodically by a single worker per dataflow running in `clusterd`.
+
+Due to the fact that there is currently no way for these dataflows to gracefully shutdown, dropping their backing replicas does not result in status updates indicating that work has been paused.
+
+A concrete example of this can be witnessed by creating in a source in some cluster, dropping the only replica in that cluster, and inspecting it status via `mz_internal.mz_source_statuses`. The current behavior is that the source is incorrectly described as `running` even though it is not possible for ingestion to occur.
+
+## Goals
+
+<!--
+Enumerate the concrete goals that are in scope for the project.
+-->
+
+- Describe how to shift the writing of status updates for sources/sinks from storage workers to the storage controller
+- How to evolve status relations to support storage clusters with multiple replicas
```

Comment:
This second goal feels pretty seperable, and I think it would be fine to sequence the changes or just punt the latter bullet to future work if you wish.